### PR TITLE
update WalletConnect v2 to 1.5.1 version, to support optionalNamespaces #6572

### LINKED ISSUE #6572: Check support for `optionalNamespaces` in WalletConnect v2 and test against labs
See thread in https://alphawalletworkspace.slack.com/archives/C03TLJYKE5D/p1679481973420459

### PR DESCRIPTION
**Pull request by oa-s**
Closes #6572

### CODE REVIEW

File: AlphaWallet/AddMultipleCustomRpc/AddMultipleCustomRpcViewController.swift [hunk 1]
```diff
@@ -192,10 +192,10 @@ extension AddCustomChain {
 
     convenience init(_ customRpc: CustomRPC, restartHandler: RestartQueueHandler, networkService: NetworkService, analytics: AnalyticsLogger) {
         let defaultDecimals = 18
-        let explorerEndpoints: [String]?
+        let explorerEndpoints: [WalletAddEthereumChainObject.ExplorerUrl]?
```
Comment (hboon) [inline AlphaWallet/AddMultipleCustomRpc/AddMultipleCustomRpcViewController.swift:195]: @oa-s is this a separate change?

Comment (oa-s) [reply AlphaWallet/AddMultipleCustomRpc/AddMultipleCustomRpcViewController.swift:195]: Nope, tester with dapp from example app, but we could extract it

**Comment by hboon** [comment]
@oa-s Ok. Buzz me if you need anything

**Comment by oa-s** [comment]
@hboon pr should be good to check. notes:
- looks like updating namespaces could throw an error on library level, not aw. (came with updated WalletConnect version) so out of this wasn't able to switch chains like before. tested with `https://interface-git-feat-add-wc2-uniswap.vercel.app/#/swap` maybe other dapps with optional namespaces would work.

tested with these daps: 
https://lab.web3modal.com/v2Extended
https://interface-git-feat-add-wc2-uniswap.vercel.app/
it works

**Review by hboon — APPROVED** [review]

**Pull request merged**